os/arch/arm/src/imxrt/imxrt_pit.c

```diff
@@ -554,24 +556,29 @@ static int imxrt_pit_ioctl(struct timer_lowerhalf_s *lower, int cmd,
 						unsigned long arg)
 {
 	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)lower;
-	int ret = ERROR;
+	uint64_t total_time;
 	uint32_t freq;
 	uint64_t ticks;
+	struct tm *l_time;
 
 	tmrvdbg("PIT IOCTL cmd: %d, arg: %ld\n", cmd, arg);
 
 	switch (cmd) {
 	case TCIOC_GETLIFETIME:
 		freq = imxrt_clock_getperclkfreq();
 		ticks = imxrt_pit_getlifetimetimercount(priv->pit->base);
-		ret = ((uint32_t)USEC_PER_SEC * (uint32_t)ticks) / freq;
-		tmrvdbg("lifetime counter %d (ms)\n", ret);
+		total_time = ((ULLONG_MAX - ticks) * USEC_PER_SEC) / freq;
+
+		total_time = total_time / USEC_PER_SEC;
+
+		l_time = localtime((const time_t *)&total_time);
```

Comment:
time_t internally uses uint32_t in TizenRT. If we use libc functions here, possible loss of data?

Reply:
The value with 32 bit counter can go beyond 100 years. I think it should be fine.

Reply:
agree